src/colourConverters.ts

```diff
@@ -0,0 +1,37 @@
+import type { ColourName } from './types/namedColours';
+import { NamedColours } from './types/namedColours';
+
+export const nameToHex = (name: ColourName): string => {
+	const hexcode = NamedColours[name];
+	return hexcode;
+};
+
+export const hexToRGB = (hex: string): number[] | null => {
+	const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
+	hex = hex.replace(
+		shorthandRegex,
+		function (m, r: string, g: string, b: string) {
+			return r + r + g + g + b + b;
+		},
+	);
+
+	const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+	return result
+		? [
+				parseInt(result[1], 16),
+				parseInt(result[2], 16),
+				parseInt(result[3], 16),
+		  ]
+		: null;
+};
+
+export const isHexCode = (hex: string): boolean =>
+	/^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.test(hex) ||
+	/^#?([a-f\d])([a-f\d])([a-f\d])$/i.test(hex);
```

Comment:
I think this can be simplified to “either 3 or 6 of [a-60-9]”.

```suggestion
	/^#?[a-f\d]{3,6}$/i.test(hex);
```

Can you tell me why the `#` is optional in this test?

Reply:
I think this is a typo actually. I adapted it from a different regex so I'll update it. 
For the suggested change: this will allow between 3 and 6 rather than 3 or 6. This would allow something like `#abcd` which I don't think is valid. That's why I split it into two tests originally. 

See below and [this link](https://regex101.com/r/rUypN5/2) for test of this.

![image](https://user-images.githubusercontent.com/9122944/102789193-b45e4180-439b-11eb-8ed5-0dc3480916fe.png)

Reply:
Ah, right, so this is how you’d do it: `^#([a-f\d]{3}|[a-f\d]{6})$`. I’ve updated [your link][101] with this example. If you prefer the javascript `||`, I still think it would be a little clearer to go with:

```typescript
	/^#[a-f\d]{6}$/i.test(hex) ||
	/^#[a-f\d]{3}$/i.test(hex) ||
```

[101]: https://regex101.com/r/rUypN5/3

Reply:
@mxdvl It's a good question. I personally find logic in regex very difficult to parse, and so thinking about it a good option might be something like :
```javascript
const longcodeRegex = /^#[a-f\d]{6}$/i.test(hex) 
const shortcodeRegex = /^#[a-f\d]{3}$/i.test(hex)

longcodeRegex.test(hex) || shortcodeRegex.test(hex)
	
```